common/isolationgroup/default-isolation-group-state.go

```diff
@@ -196,30 +293,77 @@ func isDrained(isolationGroup string, global types.IsolationGroupConfiguration,
 	return false
 }
 
-// A simple explicit deny-based isolation group implementation
-func availableIG(allIsolationGroups []string, global types.IsolationGroupConfiguration, domain types.IsolationGroupConfiguration) types.IsolationGroupConfiguration {
-	out := types.IsolationGroupConfiguration{}
-	for _, isolationGroup := range allIsolationGroups {
-		globalCfg, hasGlobalConfig := global[string(isolationGroup)]
-		domainCfg, hasDomainConfig := domain[string(isolationGroup)]
-		if hasGlobalConfig {
-			if globalCfg.State == types.IsolationGroupStateDrained {
-				continue
-			}
+// ----- Mappers -----
+func mapDynamicConfigResponse(in interface{}) (out types.IsolationGroupConfiguration, err error) {
+	if in == nil {
+		return nil, nil
+	}
+
+	dcData, ok := in.([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("failed to parse dynamic config data, unexpected format returned: %v, (%T)", in, in)
+	}
+
+	out = make(types.IsolationGroupConfiguration, len(dcData))
+	for _, v := range dcData {
+		v1, ok := v.(map[string]interface{})
+		if !ok {
+			return nil, fmt.Errorf("failed parse a dynamic config entry, %v, (got %v)", v1, v)
 		}
-		if hasDomainConfig {
-			if domainCfg.State == types.IsolationGroupStateDrained {
-				continue
-			}
+		n, okName := v1["Name"]
```

Comment:
I'm really unhappy having to hardcode these like this, but I'm covering possible serialization changes with a unit-test

Reply:
an alternative way is to serialize it and deserialize it into the type you want

Reply:
True, I didn't think of that.... I'm ok with this approach as-is for now, but if this were of significant overhead I'd give that some consideration.